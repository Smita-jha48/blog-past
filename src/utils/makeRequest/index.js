import axios from 'axios';
import { BACKEND_URL } from '../../constants/apiEndPoints';

const makeRequest = async (apiEndPoint, dynamicConfig = {}, navigate) => {
  try {
    const requestDetails = {
      baseURL: BACKEND_URL,
      url: apiEndPoint.url,
      method: apiEndPoint.method,
      ...dynamicConfig,
    };
    const { data } = await axios(requestDetails);
    return data;
  } catch (e) {
    if (navigate) {
      const errorStatus = e.response?.status;
      if (errorStatus) {
        navigate(`error/${errorStatus}`);
      } else {
        navigate('error');
      }
    }
  }
};

export default makeRequest;
