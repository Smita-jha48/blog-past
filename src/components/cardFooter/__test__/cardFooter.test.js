import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import liked from '../../../assets/Icons/heart-red.svg';
import disliked from '../../../assets/Icons/heart-black.svg';
import CardFooter from '../cardFooter';
import makeRequest from '../../../utils/makeRequest';
import clapImg from '../../../assets/Icons/clapping-grey.svg';

jest.mock('../../../utils/makeRequest');

describe('CardFooter', () => {
  const mockProps = {
    id: 1,
    like: false,
    clap: 45,
    handleLike: jest.fn(),
    handleClap: jest.fn(),
  };
  it('should render correctly', () => {
    const { asFragment } = render(<CardFooter {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render with clap image', () => {
    const { getByAltText } = render(<CardFooter {...mockProps} />);
    const imgElement = getByAltText('clapping');
    expect(imgElement.getAttribute('src')).toBe(clapImg);
  });
  it('should render with clap count', () => {
    const { getByText } = render(<CardFooter {...mockProps} />);
    const clapCountElement = getByText(mockProps.clap);
    expect(clapCountElement).toBeTruthy();
  });
  it('handleClap should be called on click', async () => {
    const { getByAltText } = render(<CardFooter {...mockProps} />);
    const imgElement = getByAltText('clapping');
    fireEvent.click(imgElement);
    await waitFor(() => {
      expect(mockProps.handleClap).toHaveBeenCalled();
    });
  });
  it('should render with heart image', async () => {
    const { getByAltText } = render(<CardFooter {...mockProps} />);
    const imgElement = getByAltText('like');
    expect(imgElement.getAttribute('src')).toBe(disliked);
  });
  it('handleLike should be called on click', async () => {
    const { getByAltText } = render(<CardFooter {...mockProps} />);
    const imgElement = getByAltText('like');
    fireEvent.click(imgElement);
    waitFor(() => {
      expect(mockProps.handleLike).toHaveBeenCalled();
    });
  });
});
