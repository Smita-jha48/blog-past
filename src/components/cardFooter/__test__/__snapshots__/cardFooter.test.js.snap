// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooter should render correctly 1`] = `
<DocumentFragment>
  <div
    class="bottom"
  >
    <div
      class="clap"
    >
      <img
        alt="clapping"
        class="reaction"
        src="clapping-grey.svg"
      />
      <p>
        45
      </p>
    </div>
    <img
      alt="like"
      class="reaction"
      src="heart-black.svg"
    />
  </div>
</DocumentFragment>
`;
